File: Backend/Data/Models/RoutePlace.cs

namespace Backend.Data.Models
{
    public class RoutePlace
    {
        public RoutePlace(string placeId, DateTime startTime, DateTime finishTime)
        {
            PlaceId = placeId;
            StartTime = startTime;
            FinishTime = finishTime;
            RouteId = 0;
            Route = null!;
            Place = null!;
        }

        public DateTime StartTime { get; set; }
        public DateTime FinishTime { get; set; }

        //Navigations Properties


        public long RouteId { get; set; }//foreignKey
        public virtual Route Route { get; set; }//foreignKey

        public string PlaceId { get; set; }//foreignKey
        public virtual Place Place { get; set; }//foreignKey
    }
}

File: lei_lds2122_grupo2-main/Frontend/src/app/models/RoutePlace.ts
/**
 * This Model represents a place in a route
 */
export class RoutePlace {

    /**
     * Construct's an instance of {@link RoutePlace}
     * 
     * @param placeId place's id
     * @param routeId route's id
     * @param startTime Start time of place visit
     * @param finishTime End time of place visit
     */
    constructor(
        public placeId: Number,
        public routeId: Number,
        public startTime: Date,
        public finishTime: Date
    ) { }
}
